assets/eip-7092/coupons/CouponMath.sol

```diff
@@ -14,10 +14,9 @@ library CouponMath {
         address _bondContract
     ) external view returns(uint256) {
         uint256 couponRate = IERC7092(_bondContract).couponRate();
-        uint256 denomination = IERC7092(_bondContract).denomination();
         uint256 principal = IERC7092(_bondContract).principalOf(_investor);
 
-        return principal * denomination * couponRate;
```

Comment:
@Pandapip1 

Here the the `principal` is in the bond currency unit, therefore we don't need to multiply the result by the `denomination`.

However, if the choice is made to express the `principal` in the `denomination` unit, then the result must be multiplied by the `denomination`.